tests/ui/simd-abi-checks.rs

```diff
@@ -0,0 +1,55 @@
+//@ only-x86_64
+//@ build-fail
+
+#![feature(avx512_target_feature)]
+#![feature(portable_simd)]
+#![allow(improper_ctypes_definitions)]
+
+use std::arch::x86_64::*;
+
+unsafe extern "C" fn f(_: __m256) {
```

Comment:
Can you add a test that involves a `repr(transparent)` wrapper around `__m256`, just to ensure we catch that as well?

Reply:
Done, that also works.